ethcore/src/blockchain/blockchain.rs

```diff
@@ -1530,8 +1530,8 @@ impl BlockChain {
 		let genesis_hash = self.genesis_hash();
 
 		// ensure data consistencly by locking everything first
-		let best_block = self.best_block.read();
 		let best_ancient_block = self.best_ancient_block.read();
+		let best_block = self.best_block.read();
```

Comment:
I don't think it's a good solution. The general rule is described here:
https://github.com/paritytech/parity-ethereum/blob/9475a2e474d9175abf05b30167bc543bd25a4e3c/ethcore/src/blockchain/blockchain.rs#L209

So simply always capture locks in the order of declaration within a struct.

Reply:
Hm, OK... But for example I wasn't aware of this comment, nor did the ones who review the PR, so I was wondering if we could find a better solution.